### PR TITLE
Sign whole metadata instead of individual claims

### PR DESCRIPTION
**Pull request by nambrot**
### Description

This PR removes the notion of signed claims and instead signs the whole metadata file. Originally I tried to make a blockchain-y construct, but it turns out to be more hassle than worth it.

### Tested

- Unit tests

### Backwards compatible

This PR is not backwards compatible as it will make all prior metadata invalid

### CODE REVIEW

File: packages/contractkit/src/identity/claims/account.test.ts [hunk 1]
```diff
@@ -68,7 +68,8 @@ testWithGanache('Account claims', (web3) => {
       IdentityMetadataWrapper.fetchFromURL = () => Promise.resolve(otherMetadata)
 
       const metadata = IdentityMetadataWrapper.fromEmpty(address)
-      signedClaim = await metadata.addClaim(
+      claim = createAccountClaim(otherAddress)
+      await metadata.addClaim(
```
Comment (timmoreton) [inline packages/contractkit/src/identity/claims/account.test.ts]: Not sure what’s happening here? Did you mean to pass in claim?

File: packages/contractkit/src/identity/claims/claim.ts [hunk 2]
```diff
@@ -103,6 +82,11 @@ export function hashOfClaim(claim: Claim) {
   return hashMessage(serializeClaim(claim))
 }
 
+export function hashOfClaims(claims: Claim[]) {
+  const hashes = claims.map(hashOfClaim)
+  return hashMessage(hashes.join(''))
```
Comment (timmoreton) [inline packages/contractkit/src/identity/claims/claim.ts:92]: Do you get a predictable ordering here?

File: packages/contractkit/src/identity/claims/account.test.ts (hunk 1 above)
Comment (nambrot) [reply packages/contractkit/src/identity/claims/account.test.ts]: I did!

File: packages/contractkit/src/identity/claims/claim.ts (hunk 2 above)
Comment (nambrot) [reply packages/contractkit/src/identity/claims/claim.ts:92]: If you are asking whether the `hashOfClaims` is yielding the same hash given an ordered `claims` parameter, then I believe so?

Comment (timmoreton) [reply packages/contractkit/src/identity/claims/claim.ts:92]: oh yeah of course SG

File: packages/cli/src/commands/account/claim-keybase.ts [hunk 3]
```diff
@@ -96,21 +92,26 @@ export default class ClaimKeybase extends ClaimCommand {
...
-  printManualInstruction(claim: SignedClaim, username: string, address: string) {
+  printManualInstruction(
+    claim: KeybaseClaim,
+    signature: string,
+    username: string,
+    address: string
+  ) {
     const fileName = proofFileName(address)
-    writeFileSync(fileName, JSON.stringify(claim))
+    writeFileSync(fileName, JSON.stringify({ claim, signature }))
```
Comment (timmoreton) [inline packages/cli/src/commands/account/claim-keybase.ts:114]: what happens if this file already exists?

**Review by timmoreton — APPROVED** [review]

File: packages/cli/src/commands/account/claim-keybase.ts (hunk 3 above)
Comment (nambrot) [reply packages/cli/src/commands/account/claim-keybase.ts:114]: We'll overwrite it

**Pull request merged**